.github/workflows/build_and_test.yml

```diff
@@ -10291,444 +10291,937 @@ jobs:
+        repository: DOCGroup/MPC
+        path: MPC
+    - name: checkout autobuild
+      uses: actions/checkout@v3
+      with:
+        repository: DOCGroup/autobuild
+        path: autobuild
+    - name: checkout ACE_TAO
+      uses: actions/checkout@v3
+      with:
+        repository: DOCGroup/ACE_TAO
+        ref: ace6tao2
+        path: ACE_TAO
+    - name: download ACE_TAO artifact
+      uses: actions/download-artifact@v3
+      with:
+        name: ACE_TAO_m10_o1d0_sec_artifact
+        path: ACE_TAO
+    - name: extract ACE_TAO artifact
+      shell: bash
+      run: |
+        cd ACE_TAO
+        tar xvfJ ACE_TAO_m10_o1d0_sec.tar.xz
+    - name: checkout OpenDDS
+      uses: actions/checkout@v3
+      with:
+        path: OpenDDS
+        submodules: true
+    - name: download OpenDDS artifact
+      uses: actions/download-artifact@v3
+      with:
+        name: build_m10_o1d0_sec_artifact
+        path: OpenDDS
+    - name: extract OpenDDS artifact
+      shell: bash
+      run: |
+        cd OpenDDS
+        tar xvfJ build_m10_o1d0_sec.tar.xz
+    - name: check build configuration
+      shell: bash
+      run: |
+        cd OpenDDS
+        . setenv.sh
+        tools/scripts/show_build_config.pl
+    - name: disable ntp
+      shell: bash
+      run: |
+        sudo systemsetup -setusingnetworktime off
+        sudo rm -rf /etc/ntp.conf
+    - name: run OpenDDS tests
+      shell: bash
+      run: |
+        cd OpenDDS
+        echo "export ACE_TEST_LOG_STUCK_STACKS=1" >> setenv.sh
+        . setenv.sh
+        mkdir ${{ github.job }}_autobuild_workspace
+        cd ${{ github.job }}_autobuild_workspace
+        echo using commit $TRIGGERING_COMMIT for SHA
+        echo $TRIGGERING_COMMIT >> ./SHA
+        cat > config.xml <<EOF
+        <autobuild>
+        <configuration>
+        <variable name="log_file" value="output.log"/>
+        <variable name="log_root" value="$DDS_ROOT/${{ github.job }}_autobuild_workspace"/>
+        <variable name="project_root" value="$DDS_ROOT"/>
+        <variable name="root" value="$DDS_ROOT/${{ github.job }}_autobuild_workspace"/>
+        <variable name="junit_xml_output" value="Tests"/>
+        </configuration>
+        <command name="log" options="on"/>
+        <command name="print_os_version"/>
+        <command name="print_env_vars"/>
+        <command name="print_ace_config"/>
+        <command name="print_cmake_version"/>
+        <command name="print_make_version"/>
+        <command name="check_compiler" options="gcc"/>
+        <command name="print_perl_version"/>
+        <command name="print_autobuild_config"/>
+        <command name="auto_run_tests" options="script_path=$ACE_ROOT/bin dir=$GITHUB_WORKSPACE/OpenDDS -l tests/core_ci_tests.lst -Config STATIC_MESSENGER -Config STATIC_COMPILER -Config STATIC_COMPILER2 -Config STATIC_UNIT_TESTS -Config XERCES3 -Config OPENDDS_SECURITY -Config CXX11 -Config RAPIDJSON -Config NO_SHMEM -Config DDS_NO_ORBSVCS -Config GH_ACTIONS_M10"/>
```

Comment:
Should we consider running a broader set of tests than the ones from `core_ci_tests.lst` for this and other test jobs. I think the build jobs have built a much broader set of tests from `dcps_tests.lst` too.

Reply:
When I attempted running  dcps_tests.lst, it seemed like it was headed for a 6 hour timeout on at least one of the builds:
https://github.com/ClaytonCalabrese/OpenDDS/runs/6082702187?check_suite_focus=true.  Looking at it again, I suppose it is possible it just takes forever to run on m11 because the other m11 build finished after almost 3 hours: https://github.com/ClaytonCalabrese/OpenDDS/runs/6082631385?check_suite_focus=true.  I'll try again with dcps_tests.lst.

Reply:
Looks like it still times out: 

https://github.com/ClaytonCalabrese/OpenDDS/runs/6134686486?check_suite_focus=true
https://github.com/ClaytonCalabrese/OpenDDS/runs/6134922703?check_suite_focus=true

Reply:
Ideally we would figure out what tests were causing the timeout and add a Config option for GH_MACOS to prevent those tests from running in GitHub Actions, but I think we should view that as a followup task and not something to accomplish in this PR. The core suite is much better than what we had previously, which was nothing.

Reply:
I agree. We can fix that and run more tests in a follow-up task.